src/libs/elektra/keyname.c

```diff
@@ -1485,71 +1502,58 @@ static size_t keyAddBaseNameInternal (Key * key, const char * baseName)
 		}
 	}
 
-	size_t newKeySize = key->keySize + escapedSize;
-	size_t newKeyUSize = key->keyUSize + unescapedSize;
-	if (test_bit (key->flags, KEY_FLAG_MMAP_KEY))
-	{
-		// key was in mmap region, clear flag and copy to malloced buffer
-		char * tmp = elektraMalloc (newKeySize);
-		memcpy (tmp, key->key, key->keySize);
-		key->key = tmp;
+	size_t newKeySize = key->keyName->keySize + escapedSize;
+	size_t newKeyUSize = key->keyName->keyUSize + unescapedSize;
 
-		tmp = elektraMalloc (newKeyUSize);
-		memcpy (tmp, key->ukey, key->keyUSize);
-		key->ukey = tmp;
+	keyDetachKeyName (key);
 
-		clear_bit (key->flags, (keyflag_t) KEY_FLAG_MMAP_KEY);
-	}
-	else
-	{
-		elektraRealloc ((void **) &key->key, newKeySize);
-		elektraRealloc ((void **) &key->ukey, newKeyUSize);
-	}
+	elektraRealloc ((void **) &key->keyName->key, newKeySize);
+	elektraRealloc ((void **) &key->keyName->ukey, newKeyUSize);
```

Comment:
This could also be combined probably. Maybe the solution in general, is adding parameters to `keyDetachKeyName` that indicate the new sizes after detaching. Then wen can realloc in one go.

Reply:
If this gets changed anyway, we probably do not optimize it here?

Reply:
What do you mean?

What I wanted to say is the `keyDetachKeyName` makes a copy, which is then immediately passed to `elektraRealloc` to resize it.  `keyDetachKeyName` could allocate the right size immediately and save one reallocation (and potential copy).

Reply:
done.

Reply:
Commenting creates notifications for me. So unless this is wanted, a simple thumbs-up or "Resolve conversation" is preferred.